### PR TITLE
doc: Split introspection documentation

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: src/main/docs/guide/ioc/introspection/introspectedAccessorsStyle.adoc [hunk 1]
```diff
@@ -0,0 +1,49 @@
+It is possible to use <<configurationPropertiesAccessorsStyle, the `@AccessorsStyle` annotation>> with `@Introspected`:
+
+[source, java]
+----
+import io.micronaut.core.annotation.AccessorsStyle;
+import io.micronaut.core.annotation.Introspected;
+
+@Introspected
+@AccessorsStyle(readPrefixes = "", writePrefixes = "") // <1>
+public class Person {
```
Comment (sdelamo) [inline src/main/docs/guide/ioc/introspection/introspectedAccessorsStyle.adoc]: @wetted move this to `test-suite`, `test-suite-groovy` and `test-suite-kotlin`

Comment (wetted) [reply src/main/docs/guide/ioc/introspection/introspectedAccessorsStyle.adoc]: @sdelamo I did this but it doesn't seem to make sense using it this way for idiomatic groovy/kotlin classes

**Comment by wetted** [comment]
When I run the `docs` task the `publishGuide` task fails with "Encountered errors while building table of contents. Aborting."

**Comment by sdelamo** [comment]
> When I run the `docs` task the `publishGuide` task fails with "Encountered errors while building table of contents. Aborting."

[Try again after ](https://github.com/micronaut-projects/micronaut-core/pull/9599/commits/e9699d123f79ed88e60795d03a0922893ad385f3)

File: src/main/docs/guide/ioc/introspection/accessKind.adoc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+By default Java introspections treat only JavaBean getters/setters or Java 16 record components as bean properties. You can however define classes with public or package protected fields in Java using the `accessKind` member of the ann:core.annotation.Introspected[] annotation:
```
Comment (wetted) [inline src/main/docs/guide/ioc/introspection/accessKind.adoc:1]:
```suggestion
By default Java introspections treat only JavaBean getters/setters or Java record components as bean properties. You can however define classes with public or package protected fields in Java using the `accessKind` member of the ann:core.annotation.Introspected[] annotation:
```

**Review by wetted — APPROVED** [review]

File: src/main/docs/guide/ioc/introspection/accessKind.adoc (hunk 2 above)
Comment (sdelamo) [reply src/main/docs/guide/ioc/introspection/accessKind.adoc:1]: specifying that records were added in java 16 is not incorrect

**Pull request merged**